subsys/bluetooth/host/shell/bt.c

```diff
@@ -597,13 +597,13 @@ static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_si
 			if (auto_connect.connect_name) {
 				auto_connect.connect_name = false;
 
-				cmd_scan_off(ctx_shell);
+				cmd_scan_off(NULL);
 
 				/* "name" is what would be in argv[0] normally */
-				cmd_scan_filter_clear_name(ctx_shell, 1, (char *[]){"name"});
+				cmd_scan_filter_clear_name(NULL, 1, (char *[]){"name"});
 
 				/* "connect" is what would be in argv[0] normally */
-				cmd_connect_le(ctx_shell, 1, (char *[]){"connect"});
+				cmd_connect_le(NULL, 1, (char *[]){"connect"});
```

Comment:
I think it's troublesome that we call these shell command handlers with a NULL shell. 

I fully understand that without the `ctx_shell`, we do not have a shell to use in those cases, but I think that we may need to refactor these calls in a way that they do not require a shell then, i.e. move the functionality from `cmd_connect_le` in a new function that is shell-less, and then call that function both here and in `cmd_connect_le`. 

Otherwise we end up in a situation where we cannot always assume that `sh != NULL`, and then we need to add `sh == NULL` checks everywhere. 

This will also remove the need for the `bt_shell_help` function

Reply:
Yeah, this looks exactly what I hoped for :)